xchange-lgo/src/main/java/org/knowm/xchange/lgo/service/LgoTradeService.java

```diff
@@ -131,61 +119,69 @@ private LgoProduct getProduct(CurrencyPair currencyPair) {
 
   @Override
   public String placeLimitOrder(LimitOrder limitOrder) throws IOException {
-    LgoPlaceOrder lgoOrder = LgoAdapters.adaptLimitOrder(limitOrder);
-    return placeOrder(lgoOrder);
+    try {
+      LgoUnencryptedOrder lgoOrder = LgoAdapters.adaptUnencryptedLimitOrder(limitOrder);
+      return placeLgoUnencryptedOrder(lgoOrder);
+    } catch (LgoException e) {
+      throw LgoErrorAdapter.adapt(e);
+    }
   }
 
-  @Override
-  public String placeMarketOrder(MarketOrder marketOrder) throws IOException {
-    LgoPlaceOrder lgoOrder = LgoAdapters.adaptMarketOrder(marketOrder);
-    return placeOrder(lgoOrder);
+  /** Place a limit order encrypting it's content. */
+  public String placeEncryptedLimitOrder(LimitOrder limitOrder) throws IOException {
+    LgoPlaceOrder lgoOrder = LgoAdapters.adaptLimitOrder(limitOrder);
+    return placeEncryptedOrder(lgoOrder);
   }
 
   @Override
-  public boolean cancelOrder(String orderId) throws IOException {
-    LgoPlaceOrder lgoOrder = LgoAdapters.adaptCancelOrder(orderId, new Date());
-    placeOrder(lgoOrder);
-    return true;
-  }
-
-  private String placeOrder(LgoPlaceOrder lgoOrder) throws IOException {
+  public String placeMarketOrder(MarketOrder marketOrder) throws IOException {
     try {
-      LgoKey lgoKey = keyService.selectKey();
-      Long ref = exchange.getNonceFactory().createValue();
-      String encryptedOrder = CryptoUtils.encryptOrder(lgoKey, lgoOrder);
-      LgoOrderSignature signature = exchange.getSignatureService().signOrder(encryptedOrder);
-      LgoEncryptedOrder lgoEncryptedOrder =
-          new LgoEncryptedOrder(lgoKey.getId(), encryptedOrder, signature, ref);
-      return placeLgoEncryptedOrder(lgoEncryptedOrder);
+      LgoUnencryptedOrder lgoOrder = LgoAdapters.adaptUnencryptedMarketOrder(marketOrder);
+      return placeLgoUnencryptedOrder(lgoOrder);
     } catch (LgoException e) {
       throw LgoErrorAdapter.adapt(e);
     }
   }
 
-  /** Place a limit order without encrypting it's content. */
-  public String placeUnencryptedLimitOrder(LimitOrder limitOrder) throws IOException {
-    try {
-      LgoUnencryptedOrder lgoOrder = LgoAdapters.adaptUnencryptedLimitOrder(limitOrder);
-      return placeLgoUnencryptedOrder(lgoOrder);
-    } catch (LgoException e) {
-      throw LgoErrorAdapter.adapt(e);
+  /** Place a market order encrypting it's content. */
+  public String placeEncryptedMarketOrder(MarketOrder marketOrder) throws IOException {
+    LgoPlaceOrder lgoOrder = LgoAdapters.adaptEncryptedMarketOrder(marketOrder);
+    return placeEncryptedOrder(lgoOrder);
+  }
+
+  @Override
+  public boolean cancelOrder(CancelOrderParams orderParams) throws IOException {
+    if (orderParams instanceof CancelOrderByIdParams) {
```

Comment:
an exception should be thrown if the required param is missing

Reply:
I couldn't find any other exchange implementations that throws in this method. Do you expect the orderId to be null ? 